src/js/components/modals/AppModalComponent.jsx

```diff
@@ -115,13 +106,9 @@ var AppModalComponent = React.createClass({
   },
 
   onInfoChange: function () {
-    this.removeInfoListeners();
-
-    if (!this.hasInfoObject()) {
-      this.setState({
-        info: InfoStore.info
-      });
-    }
+    this.setState({
```

Comment:
With once in place, I think you could just discard the request error listener, and we are fine!